x-pack/plugins/log_explorer/public/utils/convert_discover_app_state.ts

```diff
@@ -71,3 +74,60 @@ export const getDiscoverGridFromDisplayOptions = (
     return gridColumns;
   }, {}),
 });
+
+const createDiscoverPhrasesFilter = ({
+  key,
+  values,
+  negate,
+}: {
+  values: PhraseFilterValue[];
+  key: string;
+  negate?: boolean;
+}): PhrasesFilter =>
+  ({
+    meta: {
+      key,
+      type: FILTERS.PHRASES,
+      params: values,
+    },
+    query: {
+      bool: {
+        should: values.map((value) => ({ match_phrase: { [key]: value.toString() } })),
+        minimum_should_match: 1,
+      },
+    },
+  } as PhrasesFilter);
+
+const createDiscoverExistsFilter = ({
+  key,
+  negate,
+}: {
+  key: string;
+  negate?: boolean;
+}): ExistsFilter => ({
+  meta: {
+    key,
+    negate,
+    type: FILTERS.EXISTS,
+  },
+  query: { exists: { field: key } },
+});
+
+export const getDiscoverFiltersFromState = (filters: Filter[] = [], controls?: ControlOptions) => [
+  ...filters,
+  ...(controls
+    ? Object.keys(controls).map((key) =>
+        controls[key as keyof ControlOptions]?.selection.type === 'exists'
+          ? createDiscoverExistsFilter({
+              key,
+              negate: controls[key as keyof ControlOptions]?.mode === 'exclude',
+            })
+          : createDiscoverPhrasesFilter({
+              key,
+              values: (controls[key as keyof ControlOptions]?.selection as OptionsListControlOption)
+                .selectedOptions,
+              negate: controls[key as keyof ControlOptions]?.mode === 'exclude',
+            })
+      )
+    : []),
+];
```

Comment:
**suggestion:** use a single type assertion to not repeat it on each key usage:
```ts
export const getDiscoverFiltersFromState = (filters: Filter[] = [], controls?: ControlOptions) => [
  ...filters,
  ...(controls
    ? (Object.keys(controls) as Array<keyof typeof ControlOptions>).map((key) =>
        controls[key]?.selection.type === 'exists'
          ? createDiscoverExistsFilter({
              key,
              negate: controls[key]?.mode === 'exclude',
            })
          : createDiscoverPhrasesFilter({
              key,
              values: (controls[key]?.selection as OptionsListControlOption).selectedOptions,
              negate: controls[key]?.mode === 'exclude',
            })
      )
    : []),
];

```

Reply:
Addressed in [777c0fd](https://github.com/elastic/kibana/pull/173272/commits/777c0fde3964c5c91dd13790d4df0ce691f1fa52).